recipes/cppdap/all/conanfile.py

```diff
@@ -0,0 +1,99 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, copy, rm, rmdir
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+import os
+
+
+required_conan_version = ">=1.53.0"
+
+
+class PackageConan(ConanFile):
+    name = "cppdap"
+    description = "Debug Adapter Protocol SDK"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/google/cppdap"
+    topics = ("debug", "adapter", "protocol", "dap")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_json": ["jsoncpp", "nlohmann_json", "rapidjson"],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_json": "nlohmann_json",
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 11
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        if self.options.with_json == "jsoncpp":
+            self.requires("jsoncpp/1.9.5")
+        elif self.options.with_json == "rapidjson":
+            self.requires("rapidjson/cci.20220822")        
+        elif self.options.with_json == "nlohmann_json":
+            self.requires("nlohmann_json/3.11.2")
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        if self.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} can not be built as shared on Visual Studio and msvc.")
```

Comment:
if it raises for shared, why providing shared option? And why message refers to Visual Studio and msvc specifically while it raises regardless of compiler?

Reply:
FYI, shared build is allowed since https://github.com/google/cppdap/pull/116